### ISSUE TITLE
`val_losses` in each validation interval is initialized to empty and looses the previous stored validation losses.

### ISSUE THREAD

**Issue by amazloumi** [KempnerInstitute/tmrc#104]
There should be another variable like `epoch_val_losses` to keep track of all losses in each epoch to be used after step loops to calculate the average validation loss per epoch.

`val_losses` can be used for getting the average of the validation loss for the current step across val_loader's samples.
`epoch_val_losses` can gather all the above step's average losses for the current epoch and after the step loop calculate the validation loss average for the epoch.
